server/src/controllers/gmail.js

```diff
@@ -0,0 +1,72 @@
+const { google } = require('googleapis')
+const DataBaseConnectionError = require('../errors/database-connection-error')
+
+// The controller for Gmail Client
+class Gmail {
+
+    // Get an OAuthClient with configurations
+    static get_oAuthClient = () => {
+        const client_id = process.env.CLIENT_ID
+        const client_secret = process.env.CLIENT_SECRET
+        const redirect_uris = 'http://localhost:3000/gmailauth'
+
+        const oAuth2Client = new google.auth.OAuth2(
+            client_id, client_secret, redirect_uris);
+
+        return oAuth2Client
+    }
+
+    // Responds to route that gets the Authroriztion URL from google
+    static getAuthURL = async (req, res) => {
+
+        const SCOPES = ['https://www.googleapis.com/auth/gmail.readonly'];
```

Comment:
we will need also the ability to send emails

Reply:
Okay will add it in next pr